cli/azd/internal/appdetect/docker.go

```diff
@@ -1,19 +1,69 @@
 package appdetect
 
 import (
+	"bufio"
 	"io/fs"
+	"os"
 	"path/filepath"
+	"regexp"
+	"strconv"
 	"strings"
 )
 
 func detectDocker(path string, entries []fs.DirEntry) (*Docker, error) {
 	for _, entry := range entries {
 		if strings.ToLower(entry.Name()) == "dockerfile" {
+			dockerFilePath := filepath.Join(path, entry.Name())
+			exposedPorts, _ := getExposedPorts(dockerFilePath)
 			return &Docker{
-				Path: filepath.Join(path, entry.Name()),
+				Path:         dockerFilePath,
+				ExposedPorts: exposedPorts,
 			}, nil
 		}
 	}
 
 	return nil, nil
 }
+
+func getExposedPorts(dockerfilePath string) (map[int]string, error) {
```

Comment:
Let's think about breaking down this functionality differently. `getExposedPorts` right now takes the entire file, I think we could step down into a helper function that takes a line instead. I like the idea of `parsePorts(s string) ([]port, error)`.

The way I'm thinking about this: If we had to parse a different line of syntax, let's say, `RUN ...`, it wouldn't be easy to implement this additional feature without introducing additional changes. Breaking it down in the right way will also help us provide unit test coverage over the algorithm we care about: port parsing, without creating heavyweight tests that exercise the full functionality. Each test case is just a different input string, instead of a different input file.

My recommendations would be something along the following lines:

1. In `detectDocker`, let's open the file and close it for reading.
2. We read line by line using the scanner
3. If `line` has prefix `EXPOSE` -- we know it's an expose directive. We invoke `parsePorts(line[len("EXPOSE"):)` to parse the ports in the remainder of the line.

Note that `len("EXPOSE")` will be optimized by the compiler into a constant, but if we feel that the code is too verbose, we could also parse the first WORD / Directive from the line -- but I would recommend to keep the program simple at this point.

Reply:
Updated.